### PR TITLE
Travis: enable warnings and catkin_lint checker

### PR DESCRIPTION
**Pull request by rhaschke**
Enable warnings and catkin_lint checker on Travis.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # This config file for Travis CI utilizes https://github.com/ros-planning/moveit_ci/ package.
 sudo: required
-dist: trusty
+dist: xenial
```
Comment (jonbinney) [inline .travis.yml]: Were the tests running on trusty before this PR? Or am i misunderstanding what the `dist` option does?

Comment (rhaschke) [reply .travis.yml]: This simply refers to the distro ran by Travis. As we run our tests in docker, this is not really critical.
But, Travis folks indicated that they have some unresolved issues with their trusty image.

Comment (davetcoleman) [inline .travis.yml]:
```suggestion
# This is the distro ran by Travis, which may or may not be what MoveIt! is tested against within Docker
dist: xenial   
```

File: .travis.yml [hunk 2]
```diff
@@ -13,14 +13,24 @@ notifications:
       - 130s@2000.jukuin.keio.ac.jp
 env:
   global:
-    - TEST_BLACKLIST=moveit,moveit_ros,moveit_runtime,moveit_ros_perception  # mesh_filter_test fails due to broken Mesa OpenGL
+    - MOVEIT_CI_TRAVIS_TIMEOUT=85
```
Comment (davetcoleman) [inline .travis.yml]:
```suggestion
    - MOVEIT_CI_TRAVIS_TIMEOUT=85  # Travis allows us 90 min but we add a safety margin to avoid timeouts
```

**Pull request merged**